### PR TITLE
CEDS-1862 Add Choice Controller Spec + Controller Layer Infrastructure

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: test/controllers/ChoiceControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        val post = FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> Choice.ShutMUCR.value).withCSRFToken
+        val result = controller(SuccessfulAuth()).submitChoice(post)
+
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ChoiceController.displayChoiceForm().url)
+        theCacheUpserted mustBe Cache(pid, ShutMucrAnswers())
+      }
+    }
+
+    "return 400 when invalid" in {
```
Comment (ed0906) [inline test/controllers/ChoiceControllerSpec.scala]: Intentionally *not* testing the form validation here, just testing that it gets validated

**Comment by PatrykRudnicki** [comment]
And build failed because of scalafmt

File: test/controllers/ChoiceControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        contentAsHtml(result) mustBe page(Choice.form())
+      }
+
+      "existing answers" in {
+        givenTheCacheContains(Cache("pid", ArrivalAnswers()))
+
+        val result = controller(SuccessfulAuth()).displayChoiceForm(get)
+
+        status(result) mustBe Status.OK
+        contentAsHtml(result) mustBe page(Choice.form().fill(Choice.Arrival))
```
[the remark this answers is not in the document]
Comment (ed0906) [reply test/controllers/ChoiceControllerSpec.scala]: Agree that comparing the actual and expected HTML is tough.

I'll merge this then experiment with using Mocks instead

[the remark this answers is not in the document]
Comment (ed0906) [reply test/controllers/ChoiceControllerSpec.scala]: My approach to this kind of thing is usually

"write the tests/code so it reads how I want it to, then make it compile/work"

**Pull request merged**